api/v1alpha1/etcd_types.go

```diff
@@ -230,6 +230,13 @@ type EtcdConfig struct {
 	// HeartbeatDuration defines the duration for members to send heartbeats. The default value is 10s.
 	// +optional
 	HeartbeatDuration *metav1.Duration `json:"heartbeatDuration,omitempty"`
+	// ClientService defines specified annotations that should be added to the client service
+	// +optional
+	ClientService ClientServiceSpec `json:"clientService"`
```

Comment:
Shouldn't this be a pointer? ClientService should be made optional (as you have already indicated by `+ optional` comment . In additional you have already made annotations optional which is fine. If additional fields are added you can define optional/required for those fields individually.